File: ecommerce-backend/src/main/java/com/ecommerce/ecommerce_backend/dto/UserDto.java
package com.ecommerce.ecommerce_backend.dto;
import com.ecommerce.ecommerce_backend.entity.User;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class UserDto {
    private Long id;
    private String username;
    private String email;
    private String phone;
    private String fullName;
    private String avatarUrl;
    private User.Gender gender;
    private LocalDate dateOfBirth;
    private User.Role role;
}

File: ecommerce-mobile/app/src/main/java/android/app/ecommerce/data/dto/UserDto.kt
package android.app.ecommerce.data.dto

import android.app.ecommerce.data.model.Gender
import android.app.ecommerce.data.model.Role
import java.time.LocalDate

data class UserDto(
    val id: Long?,
    val username: String,
    val email: String,
    val phone: String?,
    val fullName: String,
    val avatarUrl: String?,
    val gender: Gender,
    val dateOfBirth: String,
    val role: Role
)